### PR TITLE
Install the man pages

### PR DESCRIPTION
**Pull request by lmoureaux**
## Description

This PR modifies the Makefile and setup script to install the man pages. They are installed at `$BASE/share/man`, where `$BASE` is as follows:

* `$VIRTUAL_ENV` when installed within a virtualenv
* `/usr` when installed system-wide

Man pages in the virtualenv can be accessed by setting the correct `$MANPATH`.

[outdated]
*A side-effect of this change is that I figured out how to modify the default `setuptools` install location. All scripts and macros are now installed in `$BASE/bin`. This means that `virtualenv` can take care of updating the `$PATH` (system-wide installs still need a setup script).*
[/outdated]

Another side-effect is that I corrected the license information in the package metadata. It was saying "MIT" while the `LICENSE` file always contained the text of GPLv3 (there is no way to change back to MIT without contacting every single contributor).

The setup script and developer documentation will need to be updated to use the `$MANPATH`.

### Downsides

* The `man` pages produced when the package is not installed may not be completely accurate. This is a problem for `make rpm` since it doesn't work inside a `virtualenv`.
* [outdated]*Macros and scripts are present twice in the package: under `bin/` and under `gempython/...`. I couldn't figure out why.*[/outdated]
* CI is broken :cry:

As a result of these downsides and my inability to fix them shortly, this PR is provided more as a list of commits to be cherry-picked.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

* #138
* [outdated] *Comment in `setup.py`:*
   ```
   would prefer to use this for executables, if one can control the install location
   to be user defined rather than /usr/bin
   ```
   *The preferred method (native support from `setuptools`) is now used.* [/outdated]

## How Has This Been Tested?

* The `pip` and `rpm` packages were built, the former from a `virtualenv` and the latter on `gem904daq01`.
* The `pip` package was installed; scripts and man pages are correctly installed and have the correct permissions.
* It was checked that the `rpm` package contains all files.
* The `man` pages built in each environment were (quickly) checked.

Since I don't have access to a `rpm`-based machine and I don't want to install one, I couldn't check that the `rpm` package installs cleanly.

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: pkg/installrpm.sh
Comment (jsturdy) [inline pkg/installrpm.sh]: This was also explicitly avoided in the initial setup, as in future versions this will be replaced with system site-packages location macros provided by `rpm` (currently only works for `cc7`)

File: pkg/setup.cfg
Comment (jsturdy) [inline pkg/setup.cfg:7]: License changes *must* be discussed

File: pkg/setup.cfg
Comment (jsturdy) [inline pkg/setup.cfg:23]: Same

File: pkg/setup.py
Comment (jsturdy) [inline pkg/setup.py:63]: Also, prefer consistent string formatting (avoid `%` formatting in favour of `.format()` formatting) for all current and future contributions, `CONTRIBUTING.md` will be updated to reflect this and a few other changes

**Review by jsturdy — CHANGES_REQUESTED** [review]
There are far too many changes in this PR that should have been discussed in #138 before being implemented

> A side-effect of this change is that I figured out how to modify the default setuptools install location. All scripts and macros are now installed in $BASE/bin. This means that virtualenv can take care of updating the $PATH (system-wide installs still need a setup script).

This was by design, avoiding putting things in `macros` into `/opt/cmsgemos/bin`

> Another side-effect is that I corrected the license information in the package metadata. It was saying "MIT" while the LICENSE file always contained the text of GPLv3 (there is no way to change back to MIT without contacting every single contributor).

This is probably the discussion that I mention inline.  I would have to go back in time to figure out why a given license was (probably arbitrarily) chosen for this project, and probably raise it as a discussion, (contacting contributors on this project is pretty trivial)

File: pkg/setup.py
Comment (bdorney) [reply pkg/setup.py:63]: For my own ignorance, why is `.format()` viewed as superior?

File: pkg/setup.cfg
Comment (lmoureaux) [reply pkg/setup.cfg:7]: This is not a license change, this is reflecting the fact that the current license *is* GPLv3. If you want to contact all copyright holders, fine; but until you get answers the license remains GPLv3.

File: pkg/installrpm.sh
Comment (lmoureaux) [reply pkg/installrpm.sh]: > system site-packages location macros provided by `rpm`

Could you provide more details about this? A link to the `rpm` doc about this would be great

**Comment by lmoureaux** [comment]
> avoiding putting things in macros into `/opt/cmsgemos/bin`

May I ask what's the rationale for the difference between `scripts` and `macros` in this project?

File: pkg/setup.py
Comment (jsturdy) [reply pkg/setup.py:63]: I don't know about *superior*... it is the standard that was introduced to eventually replace `%` formatting, though I suspect that `%` formatting will stick around for quite some time...
Both have a few advantages, though I think `.format()` has more.
The primary thing here (in this comment) is maintaining consistency

**Comment by lmoureaux** [comment]
> I would have to go back in time to figure out why a given license was (probably arbitrarily) chosen for this project 

Commit 3d58f5d2 by... @jsturdy. The commit message ("Initial commit") isn't helpful in telling how the license was chosen.

FYI:
* CMSSW uses [Apache 2.0](http://www.apache.org/licenses/LICENSE-2.0).
* XDAQ is under a 3-clause [BSD-style license](https://svnweb.cern.ch/trac/cmsos/browser/trunk/LICENSE)

**Comment by jsturdy** [comment]
> May I ask what's the rationale for the difference between scripts and macros in this project?

Mainly, it is to isolate the key executable parts (analyzing scans, which should go in `/opt/cmsgemos/bin`) from random plotting and style macros (can still be on the `PATH` for ease of use, provided `PATH` is modified to include the `macros` directory)

Further down the road, much of this will be replaced with `console_scripts` and `entry_points`, but that is a minor priority

**Comment by lmoureaux** [comment]
Removed the controversial move to a more standard behavior using standard tools; the `rpm` package will now install `man` pages in `/usr/share`. Updated the first post accordingly.

**Comment by lmoureaux** [comment]
Comments have been addressed and I'm waiting for another review.

**Review by jsturdy — APPROVED** [review]
RPM requirements:
```bash
rpm -q --requires -p rpm/gempython_gemplotting-1.0.0-0.0.6.30.dev5.087aeffgit.centos7.python2.7.noarch.rpm
/bin/env
/bin/sh
/usr/bin/env
cmsgemos_gempython >= 0.3.1
config(gempython_gemplotting) = 1.0.0-0.0.6.30.dev5.087aeffgit.centos7.python2.7
numpy >= 1.7
python >= 2.6
python(abi) = 2.7
root_numpy >= 4.7
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PartialHardlinkSets) <= 4.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
tabulate >= 0.4.4
rpmlib(PayloadIsXz) <= 5.2-1
```

* looks like some mixed shebangs are used, these should be cleaned up in the future
* also in the future, would like to be able to provide these to `bdist_rpm` based on the `python` version, but I believe this would be a significant redesign, moving the majority of the `setup.cfg` stuff into functions imported in `setup.py`

**Pull request merged**
